pkg/shell/base_index.js

```diff
@@ -613,6 +627,51 @@
         return false;
     };
 
+    function setDarkMode(documentElement, dark_mode) {
+        if (dark_mode) {
+            documentElement.classList.add('pf-theme-dark');
+        } else {
+            documentElement.classList.remove('pf-theme-dark');
+        }
+    }
+
+    function updateFrames(dark_mode) {
+        setDarkMode(document.documentElement, dark_mode);
+        Object.values(self.frames.iframes).forEach(machine => {
+            Object.values(machine).forEach(frame => {
+                setDarkMode(frame.contentDocument.documentElement, dark_mode);
+            });
+        });
+    }
+
+    /* dark mode */
+    window.addEventListener("cockpit-style", event => {
+        const style = event.detail.style;
+        localStorage.setItem("shell:style", style);
+
+        if ((window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches && style === "auto") || style === "dark") {
+            self.dark_mode = true;
+            updateFrames(true);
+        } else {
+            self.dark_mode = false;
+            updateFrames(false);
+        }
+    });
+
+    window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', event => {
+        const style = localStorage.getItem('shell:style') || 'auto';
+        // If a user set's an explicit theme, ignore system changes.
+        if (style !== "auto")
+            return;
+
+        if (event.matches) {
+            self.dark_mode = true;
+            updateFrames(true);
+        } else {
+            self.dark_mode = false;
+            updateFrames(false);
+        }
+    });
```

Comment:
These 8 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17914-20221201-104014-14c78ccb-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)